### ISSUE TITLE
az artifacts universal download fails on centos

### ISSUE THREAD

**Issue by harshg0910** [Azure/azure-cli-extensions#623]
### Extension name (the extension in question)
azure-devops

### Description of issue (in as much detail as possible)
OS: CentOS Linux release 7.6.1810 (Core)
Installed using "pip install azure-cli"
azure-cli version: 2.0.61

I'm able to login to the cli but when I'm trying to use 
```
$az artifacts universal download [arguments]

The command failed with an unexpected error. Here is the traceback:

'module' object has no attribute 'uuid4'
Traceback (most recent call last):
  File "/usr/lib64/az/lib/python2.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib64/az/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 351, in execute
    raise ex
AttributeError: 'module' object has no attribute 'uuid4'
```
-----

**Comment by psericw** [comment]
I'm seeing the same thing...

OS: CentOS Linux release 7.6.1810 (Core) 
Installed AZ CLI using: $ sudo yum install azure-cli
AZ CLI version: 2.0.61
Installed devops extension using: $ az extension add --name azure-devops
azure-devops extension version: 0.4.0

However, using the exact same versions of AZ CLI and azure-devops extension, but on OS Ubuntu 18.04.2 LTS, this problem does not occur.

**Comment by mamboking** [comment]
I'm also seeing this using:

Linux-3.10.0-957.12.1.el7.x86_64-x86_64-with-centos-7.6.1810-Core
Python 2.7.5
Shell: bash

azure-cli 2.0.64

Extensions:
azure-devops 0.8.0

**Comment by justinttl** [comment]
Same here. 
![image](https://user-images.githubusercontent.com/9387665/58280252-30cb1180-7d6e-11e9-9e04-cb96518eb920.png)

Also running Centos7

**Comment by harshg0910** [comment]
As a temporary workaround, install and use Azure CLI in python 3 virtual environment. I have made 3 scripts: 
1. **install**: creates a virtual environment and install azure cli using pip
2. **download**: It activates that virtual environment, download the package and deactivates the env
3. **upload**: Similar to download

**Comment by geverghe** [comment]
@elbatk - Could you have a look into this?

**Comment by AdamClarkRock** [comment]
I managed to get it working on ubuntu 18.04, but it failed in my WSL ubuntu with the same exception stack as above

**Comment by tkrick** [comment]
@harshg0910 et al, have you tried running these steps? https://github.com/dotnet/core/blob/master/Documentation/build-and-install-rhel6-prerequisites.md

universal packages cli extension depends on a .net core app and apparently some of the distros runtime library mismatches with what .net core needs. I'm just guessing because I notice uuid is the issue and these instructions mention a mismatch with: libuuid

**Comment by harshg0910** [comment]
I think the libuuid is there as I was able to run it properly with using python virtualenv. I feel the problem maybe python version. Python 2.7 is not supported by azure CLI and it recommends upgrading system wide python to make it work (which may not be feasible everywhere for legacy dependencies). 

I installed python 3.6 as python3 and used it's pip to install the CLI, but apparently the internally it refers to system python (which is weird as the python used is inside az directory /usr/lib64/az/lib/python2.7) and breaks. So, using virtualenv will override system python and will work.
